src/System.Linq/src/System/Linq/Partition.cs

```diff
@@ -42,22 +42,18 @@ internal interface IPartition<TElement> : IIListProvider<TElement>
 
         IPartition<TElement> Take(int count);
 
-        TElement ElementAt(int index);
+        TElement TryGetElementAt(int index, out bool found);
 
-        TElement ElementAtOrDefault(int index);
+        TElement TryGetFirst(out bool found);
 
-        TElement First();
-
-        TElement FirstOrDefault();
-
-        TElement Last();
-
-        TElement LastOrDefault();
+        TElement TryGetLast(out bool found);
```

Comment:
For some reason the analysis part of running a performance test is failing for me, so I get an empty HTML file. The .etl file is also causing Performance Analyzer to crash (which may be the cause of the empty HTML). I can see total times in the XML though.

It does seem to be a vindication of @VSadov's advice, but a slight one. Cases that result in a throw (due to not found when that should throw) come out slightly slower, but all the rest faster. It occurs to me that it's possible that the matter of which value is in which register upon returning from a method is also a factor here.

I decided not to test cases where the `out` addressed a field, since that case doesn't come up in any of the calls here.